### PR TITLE
Bug/dsg 1706/fix sonarqube code smells duplication

### PR DESCRIPTION
**Pull request by kevincheng0**
# Basics

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [x] Code style changes (formatting, local variables)
- [x] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?
Fixing code smells reported on SonarQube. There are 3 false positives with the storybook files saying that variables aren't being used, but they are used in the HTML templates and SonarQube didn't detect that. Another false positive with casting that our linting rules won't allow to be changed.

Added ESLint rule `no-nested-ternary` to match SonarQube analysis.

False positives:
https://sonarqube.dsgov.demo.nuvalence.io/project/issues?issues=AYpsh0rwWkBWa3O5mJjC&open=AYpsh0rwWkBWa3O5mJjC&id=dsgov-web
https://sonarqube.dsgov.demo.nuvalence.io/project/issues?issues=AYgNHdCAuegd7MY7bX9D&open=AYgNHdCAuegd7MY7bX9D&id=dsgov-web
https://sonarqube.dsgov.demo.nuvalence.io/project/issues?issues=AYgNHdCAuegd7MY7bX9E&open=AYgNHdCAuegd7MY7bX9E&id=dsgov-web
https://sonarqube.dsgov.demo.nuvalence.io/project/issues?issues=AYgNHdFAuegd7MY7bX9H&open=AYgNHdFAuegd7MY7bX9H&id=dsgov-web

image

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [x] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/agency/feature/dashboard/src/lib/components/events-log/event-types/transaction-data-changed/transaction-data-changed.component.ts [hunk 1]
```diff
@@ -49,46 +49,45 @@ export class TransactionDataChangedComponent implements OnInit {
 
   private handleTransactionDataUpdatedEvent(event: AuditEventModel) {
     const { newState, oldState } = event.eventData;
+    if (!newState || !oldState) return; // Nothing happens if either is undefined
```
Comment (davidyoung77) [inline libs/agency/feature/dashboard/src/lib/components/events-log/event-types/transaction-data-changed/transaction-data-changed.component.ts:52]: check to see if we can update our linting rules to match

File: libs/shared/feature/authentication/src/lib/components/authentication-email/authentication-email.component.ts [hunk 2]
```diff
@@ -116,7 +116,10 @@ export class AuthenticationEmailComponent extends AuthenticationBaseDirective im
 
     this.submitStatus$ = combineLatest([this.formGroup.statusChanges.pipe(startWith(null)), this._submitStatus.pipe(startWith(null))]).pipe(
       map(([formStatus, submitStatus]) => {
-        const status: SubmitStatusType = !formStatus || formStatus === 'INVALID' ? 'invalid' : submitStatus ? 'loading' : 'valid';
+        let status: SubmitStatusType = 'invalid';
```
Comment (davidyoung77) [inline libs/shared/feature/authentication/src/lib/components/authentication-email/authentication-email.component.ts:119]: check to see if we can update our linting rules to match

File: libs/shared/feature/authentication/src/lib/adapters/firebase/firebase-authentication.adapter.ts [hunk 3]
```diff
@@ -78,10 +79,10 @@ export class FirebaseAuthenticationAdapter implements AuthenticationAdapter {
         user: credential.user,
       })),
       catchError((error: firebase.FirebaseError) => {
-        throw {
+        throw new AuthenticationError({
```
Comment (davidyoung77) [inline libs/shared/feature/authentication/src/lib/adapters/firebase/firebase-authentication.adapter.ts:82]: I'd like to see if we can handle this better

**Review by davidyoung77 — COMMENTED** [review]
Thanks I have a couple comments

File: libs/shared/feature/authentication/src/lib/adapters/firebase/firebase-authentication.adapter.ts (hunk 3 above)
Comment (kevincheng0) [reply libs/shared/feature/authentication/src/lib/adapters/firebase/firebase-authentication.adapter.ts]: Updated with RxJS throwError and new Error() as appropriate

File: libs/shared/feature/authentication/src/lib/components/authentication-email/authentication-email.component.ts (hunk 2 above)
Comment (kevincheng0) [reply libs/shared/feature/authentication/src/lib/components/authentication-email/authentication-email.component.ts:119]: yes, there is a `no-nested-ternary` rule for ESLint!

File: libs/agency/feature/dashboard/src/lib/components/events-log/event-types/transaction-data-changed/transaction-data-changed.component.ts (hunk 1 above)
Comment (kevincheng0) [reply libs/agency/feature/dashboard/src/lib/components/events-log/event-types/transaction-data-changed/transaction-data-changed.component.ts:52]: unfortunately I could not find any linting rules to match this

**Review by davidyoung77 — APPROVED** [review]
Nice work lets sync and look at SQ together after this is merged

**Pull request merged**